### PR TITLE
gcp - Cloud Storage initial setup with nested resources

### PR DESCRIPTION
**Pull request by Dkhanas**
Added initial classes with query and get for BucketAccessControls, DefaultObjectAccessControls, Objects , ObjectAccessControls  and Notifications data resource in Cloud Storage service

### CODE REVIEW

**Comment by aleonidex** [comment]
**Use case:** we can set up Custodian to filter objects of certain configurable attributes (name, storage class, creation date, etc) and do something about them (upon enabling of such actions). In other words, Custodian can be used to control access to objects with sensitive data and notify about major changes in it.

File: tools/c7n_gcp/tests/test_gcp_storage.py [hunk 1]
```diff
@@ -31,15 +31,158 @@ def test_bucket_query(self):
...
+                              },
                              session_factory=factory)
-        bucket = p.resource_manager.get_resource({
+        exec_mode = p.get_execution_mode()
+        event = event_data('cs-bucket-update.json')
+        instance = exec_mode.run(event, None)
+        self.assertEqual(instance[0]['bucket'], bucket_name)
+
+
+class DefaultObjectAccessControlTest(BaseTest):
```
Comment (jondavidi) [inline tools/c7n_gcp/tests/test_gcp_storage.py]: add `Bucket` to class name to be consistent

File: tools/c7n_gcp/c7n_gcp/resources/storage.py [hunk 2]
```diff
@@ -25,8 +30,140 @@ class resource_type(TypeInfo):
         scope = 'project'
         enum_spec = ('list', 'items[]', {'projection': 'full'})
         id = 'name'
+        get_requires_event = True
+
+        @staticmethod
+        def get(client, event):
+            if 'resource' in event.keys():
```
[the remark this answers is not in the document]
Comment (Dkhanas) [reply tools/c7n_gcp/c7n_gcp/resources/storage.py]: Updated

File: tools/c7n_gcp/c7n_gcp/resources/storage.py [hunk 3]
```diff
@@ -25,8 +30,140 @@ class resource_type(TypeInfo):
...
+                entity = '-'.join(entity.split(':'))
+            return client.execute_command(
+                'get', {
+                    'bucket': jmespath.search('resource.labels.bucket_name', event),
+                    'entity': entity
+                })
+
+
+@resources.register('bucket-default-object-access-control')
+class DefaultObjectAccessControl(ChildResourceManager):
```
[the remark this answers is not in the document]
Comment (Dkhanas) [reply tools/c7n_gcp/c7n_gcp/resources/storage.py]: Updated

File: tools/c7n_gcp/tests/test_gcp_storage.py (hunk 1 above)
Comment (Dkhanas) [reply tools/c7n_gcp/tests/test_gcp_storage.py]: Updated

**Comment by jondavidi** [comment]
@Dkhanas did you forget to push the latest commits to this PR? I'm not seeing the changes you said you made

**Comment by Dkhanas** [comment]
@jondavidi Yes, i forgot push changes. Now updated

**Comment by aleonidex** [comment]
@jondavidi the recent changes are now in, please review

**Comment by jondavidi** [comment]
@Dkhanas the versioning history with this PR is out of whack. Please take a look at it and see the problems, the most evident being the 116 count of files modified which I do not believe are most of your changes. I think it's something to do with the way you've merged and force-pushed (which I've seen a lot of from the EPAM team and is a broader issue because it causes some problems with 
version history). Observe how the PR conversation itself says that Kapil and others have added to this PR.

I would suggest potentially just raising a new PR with your changes. probably esasier than trying to fix this one

**Pull request closed without merging**